src/status.c

```diff
@@ -0,0 +1,28 @@
+#include "status.h"
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+static Status global_status = {.code = STATUS_CODE_OK, .caller = NULL, .message = "" };
+
+bool prv_status_create(const StatusCode code, const char* caller, const char* message) {
+  if (code >= NUM_STATUS_CODE) {
```

Comment:
I feel like error handling in our error handler is probably overkill. :P
But really, I don't think there's much use in setting the invalid input to more invalid input.

For this function, I think we might as well just return the enum as is. I think it's easier to just process the enum as a return value instead of having to check the return value and then call a function to further handle the error. For example, if we only care if a specific error occurred, the logical step would be to ignore the return value and just get the last status through the function. I do see the benefit in keeping the original return code throughout the callstack, but I think our default behavior would be fine as returning the code as is.